### PR TITLE
Add observe method for excluding UIView or UIScrollView from swipe gesture recognizing

### PR DESCRIPTION
**Pull request by alexstich**

### CODE REVIEW

File: Sources/SwipeTransition/SwipeBack/SwipeBackController.swift [hunk 1]
```diff
@@ -58,14 +58,30 @@ public final class SwipeBackController: NSObject {
...
+        shouldBeginSwipeTransition = shouldBeginSwipe
+    }
+    
+    public func observeViewController(_ scrollView: UIScrollView?, shouldBeginSwipe: @escaping () -> Bool) {
+        if scrollView != nil {
+            scrollView!.panGestureRecognizer.require(toFail: panGestureRecognizer)
+            context.pageViewControllerGestureRecognizer = scrollView!.panGestureRecognizer
+            shouldBeginSwipeTransition = shouldBeginSwipe
+        }
+    }
```
Comment (tattn) [inline Sources/SwipeTransition/SwipeBack/SwipeBackController.swift:77]: If scrollView is nil, this method does nothing, so it is preferable not to use Optional at the argument.
```suggestion
    public func observeViewController(_ scrollView: UIScrollView, shouldBeginSwipe: @escaping () -> Bool) {
        scrollView.panGestureRecognizer.require(toFail: panGestureRecognizer)
        context.pageViewControllerGestureRecognizer = scrollView.panGestureRecognizer
        shouldBeginSwipeTransition = shouldBeginSwipe
    }
```

File: Sources/SwipeTransition/SwipeBack/SwipeBackController.swift [hunk 2]
```diff
@@ -58,14 +58,30 @@ public final class SwipeBackController: NSObject {
...
+    
+    public func observeViewController(view: UIView?, shouldBeginSwipe: @escaping () -> Bool) {
+        if view != nil {
+            view!.gestureRecognizers?.first?.require(toFail: panGestureRecognizer)
+            context.pageViewControllerGestureRecognizer = view!.gestureRecognizers?.first
+            shouldBeginSwipeTransition = shouldBeginSwipe
+        }
```
Comment (tattn) [inline Sources/SwipeTransition/SwipeBack/SwipeBackController.swift:84]: This is the same as the scrollView, too.
```suggestion
    public func observeViewController(view: UIView, shouldBeginSwipe: @escaping () -> Bool) {
        view.gestureRecognizers?.first?.require(toFail: panGestureRecognizer)
        context.pageViewControllerGestureRecognizer = view.gestureRecognizers?.first
        shouldBeginSwipeTransition = shouldBeginSwipe
```

**Review by tattn — CHANGES_REQUESTED** [review]
Thanks for the PR!

I'd like to merge these changes but would like to change the interface a bit.

**Comment by alexstich** [comment]
I wanted to add an opportunity to add a few views in an array of excluded views that don't influence swipe transition. But couldn't do this. This pull request allows adding only one UIScrollView or UIView or UIPageViewController. It is not the best solution. But it help me a few times.

File: Sources/SwipeTransition/SwipeBack/SwipeBackController.swift [hunk 3]
```diff
@@ -58,14 +58,26 @@ public final class SwipeBackController: NSObject {
         context.navigationControllerDelegateProxy = NavigationControllerDelegateProxy(delegates: [self] + (delegate.map { [$0] } ?? []) )
     }
 
-    public func observePageViewController(_ pageViewController: UIPageViewController, isFirstPage: @escaping () -> Bool) {
+    public func observePageViewController(_ pageViewController: UIViewController, shouldBeginSwipe: @escaping (UIGestureRecognizer?) -> Bool) {
```
Comment (tattn) [inline Sources/SwipeTransition/SwipeBack/SwipeBackController.swift:61]: I want to change the name to "observe" to unify the naming convention.
```suggestion
    public func observe(viewController: UIViewController, shouldBeginSwipe: @escaping (UIGestureRecognizer) -> Bool) {
```

File: Sources/SwipeTransition/SwipeBack/SwipeBackController.swift [hunk 4]
```diff
@@ -58,14 +58,26 @@ public final class SwipeBackController: NSObject {
...
             .compactMap { $0 as? UIScrollView }
             .first
         scrollView?.panGestureRecognizer.require(toFail: panGestureRecognizer)
-        context.pageViewControllerPanGestureRecognizer = scrollView?.panGestureRecognizer
-        isFirstPageOfPageViewController = isFirstPage
+        context.pageViewControllerGestureRecognizer = scrollView?.panGestureRecognizer
+        shouldBeginSwipeTransition = shouldBeginSwipe
+    }
+    
+    public func observeViewController(_ scrollView: UIScrollView, shouldBeginSwipe: @escaping (UIGestureRecognizer?) -> Bool) {
```
Comment (tattn) [inline Sources/SwipeTransition/SwipeBack/SwipeBackController.swift:71]:
```suggestion
    public func observe(scrollView: UIScrollView, shouldBeginSwipe: @escaping (UIGestureRecognizer) -> Bool) {
```

File: Sources/SwipeTransition/SwipeBack/SwipeBackController.swift [hunk 5]
```diff
@@ -58,14 +58,26 @@ public final class SwipeBackController: NSObject {
...
+        scrollView.panGestureRecognizer.require(toFail: panGestureRecognizer)
+        context.pageViewControllerGestureRecognizer = scrollView.panGestureRecognizer
+        shouldBeginSwipeTransition = shouldBeginSwipe
+    }
+    
+    public func observeViewController(view: UIView, shouldBeginSwipe: @escaping (UIGestureRecognizer?) -> Bool) {
```
Comment (tattn) [inline Sources/SwipeTransition/SwipeBack/SwipeBackController.swift:77]:
```suggestion
    public func observe(view: UIView, shouldBeginSwipe: @escaping (UIGestureRecognizer) -> Bool) {
```

File: Sources/SwipeTransition/SwipeBack/SwipeBackController.swift [hunk 6]
```diff
@@ -12,7 +12,7 @@ import UIKit
 public final class SwipeBackController: NSObject {
     public var onStartTransition: ((UIViewControllerContextTransitioning) -> Void)?
     public var onFinishTransition: ((UIViewControllerContextTransitioning) -> Void)?
-    private var isFirstPageOfPageViewController: (() -> Bool)?
+    private var shouldBeginSwipeTransition: ((UIGestureRecognizer?) -> Bool)?
```
Comment (tattn) [inline Sources/SwipeTransition/SwipeBack/SwipeBackController.swift:15]: UIGestureRecognizer does not seem to need to be Optional.
```suggestion
    private var shouldBeginSwipeTransition: ((UIGestureRecognizer) -> Bool)?
```

**Review by tattn — CHANGES_REQUESTED** [review]
This PR is great!
Finally, I'd like to make these changes for uniformity of method naming conventions.
And then I'll merge this PR.

**Comment by alexstich** [comment]
After adding gesture to observer callback I use this method like this: 

`        self.vc?.navigationController?.swipeBack?.observe(view: observed_view, shouldBeginSwipe:
            { [weak self] gesture in
            
                guard let self = self else { return true }
            
                let frame_ = self.observed_view.frame
                let location = gesture.location(in: self.observed_view)
            
                if location.x < 0 || location.y < 0 || location.x > frame_.width || location.y > frame_.height {
                    return true
                }
            
                return false
            }
        )`

**Review by tattn — APPROVED** [review]
Thanks for the great improvements!

**Pull request merged**